new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Pod.java

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.assertions.impl;
+
+import java.util.ArrayList;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1Pod;
+import io.kubernetes.client.openapi.models.V1PodList;
+import org.awaitility.core.ConditionFactory;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class Pod {
+
+  // reusable condition factory
+  private static ConditionFactory retry
+      = with().pollInterval(5, SECONDS).atMost(10, MINUTES).await();
+
+  /**
+   * Check the pods in the given namespace are restarted in a rolling fashion.
+   * Waits until all pods are restarted, for upto 10 minutes.
+   *
+   * @param domainUid UID of the WebLogic domain
+   * @param namespace name of the namespace in which to check for the pods restart sequence
+   * @return true if pods in the namespace are restarted in a rolling fashion otherwise false
+   * @throws ApiException when Kubernetes cluster query fails
+   * @throws InterruptedException when pod status check threads are interrupted
+   * @throws ExecutionException when pod status checks times out
+   * @throws TimeoutException when waiting for the threads times out
+   */
+  public static boolean isARollingRestart(String domainUid, String namespace)
+      throws ApiException, InterruptedException, ExecutionException, TimeoutException {
+
+    // query cluster and get pods from the namespace
+    String labelSelectors = "weblogic.serverName";
+    V1PodList listPods = Kubernetes.listPods(namespace, labelSelectors);
+    ArrayList<String> podNames = new ArrayList<>();
+
+    //return if no pods are found
+    if (listPods.getItems().isEmpty()) {
+      logger.severe("No pods found in namespace {0}", namespace);
+      return false;
+    } else {
+      logger.info("WebLogic pods found in namespace {0}", namespace);
+      for (V1Pod item : listPods.getItems()) {
+        logger.info(item.getMetadata().getName());
+        podNames.add(item.getMetadata().getName());
+      }
+    }
+
+    // check the pods termination status in concurrent thread
```

Comment:
check that all of the pods are in termination status in a set of concurrent threads

Reply:
checking the pod termination status is now sequential according to the order of pods in the list. Also while checking for termination status of a  given pod , the method will check no other pods are terminating along with it.